src/explore-education-statistics-admin/src/pages/release/content/components/__tests__/EditableKeyStatText.test.tsx

```diff
@@ -167,4 +171,160 @@ describe('EditableKeyStatText', () => {
       screen.queryByRole('button', { name: /Remove/ }),
     ).not.toBeInTheDocument();
   });
+
+  describe('when editing', () => {
+    test('renders edit form correctly', async () => {
+      render(
+        <EditableKeyStatText keyStat={keyStatText} isEditing onSubmit={noop} />,
+      );
+
+      await waitFor(() => {
+        expect(screen.getByTestId('keyStat-title')).toHaveTextContent(
+          'Text title',
+        );
+        expect(screen.getByTestId('keyStat-statistic')).toHaveTextContent(
+          'Over 9000',
+        );
+        expect(screen.getByTestId('keyStat-trend')).toHaveTextContent(
+          'Text trend',
+        );
+        expect(
+          screen.getByRole('button', {
+            name: 'Text guidance title',
+          }),
+        ).toBeInTheDocument();
+
+        expect(screen.getByTestId('keyStat-guidanceText')).toHaveTextContent(
+          'Text guidance text',
+        );
+
+        expect(
+          screen.getByRole('button', {
+            name: 'Edit key statistic: Text title',
+          }),
+        ).toBeInTheDocument();
+
+        expect(
+          screen.queryByRole('button', { name: /Remove/ }),
+        ).not.toBeInTheDocument();
+      });
+    });
+
+    test('submitting edit form calls `onSubmit` handler with updated values', async () => {
+      const onSubmit = jest.fn();
+
+      render(
+        <EditableKeyStatText
+          keyStat={keyStatText}
+          isEditing
+          onSubmit={onSubmit}
+        />,
+      );
+
+      await waitFor(() => {
+        expect(screen.getByText(/Edit/)).toBeInTheDocument();
+      });
+
+      userEvent.click(
+        screen.getByRole('button', {
+          name: 'Edit key statistic: Text title',
+        }),
+      );
+
+      await waitFor(() => {
+        expect(screen.getByLabelText('Title')).toBeInTheDocument();
+      });
+
+      await userEvent.clear(screen.getByLabelText('Title'));
+      await userEvent.type(screen.getByLabelText('Title'), 'New title');
+
+      await userEvent.clear(screen.getByLabelText('Statistic'));
+      await userEvent.type(screen.getByLabelText('Statistic'), 'New statistic');
+
+      await userEvent.clear(screen.getByLabelText('Trend'));
+      await userEvent.type(screen.getByLabelText('Trend'), 'New trend');
+
+      await userEvent.clear(screen.getByLabelText('Guidance title'));
+      await userEvent.type(
+        screen.getByLabelText('Guidance title'),
+        '  New guidance title  ', // Whitespace should be trimmed
```

Comment:
Rather than have this as a comment it'd be better to have a specific test for trimming whitespace.

Reply:
Also added the same additional test to `EditableKeyStatDataBlock.test.tsx`